internal/gateway/statistics/statistics.go

```diff
@@ -20,33 +21,41 @@ type Counters struct {
 // TODO: concurrent access
 type StatisticsMgr struct {
 	leaseCounters map[string]Counters
+	rwLock        sync.RWMutex
 }
 
-func NewStatisticsMgr() (*StatisticsMgr, error) {
-	return &StatisticsMgr{leaseCounters: make(map[string]Counters)}, nil
+func NewStatisticsMgr() *StatisticsMgr {
+	return &StatisticsMgr{leaseCounters: make(map[string]Counters)}
 }
 
-func (m *StatisticsMgr) AddLease(leasePath string, counters Counters) error {
-	m.leaseCounters[leasePath] = counters
+func (m *StatisticsMgr) CreateLease(leasePath string) error {
+	m.rwLock.Lock()
+	if _, ex := m.leaseCounters[leasePath]; ex {
+		m.rwLock.UnLock()
```

Comment:
much more idiomatic to use defer as soon as you lock the mutex.

Reply:
ah, thanks!